src/web/web_actions.nit

```diff
@@ -0,0 +1,163 @@
+class SearchAction
+	super NitAction
+
+	# Model to explore and render.
+	var model: Model
+
+	# TODO handle more than full namespaces.
+	redef fun answer(request, url) do
+		var namespace = request.param("namespace")
+		if namespace == null or namespace.is_empty then
+			return render_error(400, "Missing :namespace.")
+		end
+		var mentities = model.collect_by_namespace(namespace)
+		if request.bool_arg("json") or else false then
+			var json = new JsonArray
+			for mentity in mentities do
+				json.add mentity.to_json
+			end
+			return render_json(json)
+		end
+		var view = new HtmlResultPage(namespace, mentities)
+		return render_view(view)
+	end
+end
+
+# Display a MEntity source code.
+class CodeAction
+	super NitAction
+
+	# Model to explore and render.
+	var model: Model
+
+	# Modelbuilder used to access sources.
+	var modelbuilder: ModelBuilder
+
+	# TODO handle more than full namespaces.
+	redef fun answer(request, url) do
+		var namespace = request.param("namespace")
+		if namespace == null or namespace.is_empty then
+			return render_error(400, "Missing :namespace.")
+		end
+		var mentities = model.collect_by_namespace(namespace)
+		if mentities.is_empty then
+			return render_error(404, "No mentity matching this namespace.")
+		end
+		var view = new HtmlSourcePage(modelbuilder, mentities.first)
+		return render_view(view)
+	end
+end
+
+# Display the doc of a MEntity.
+class DocAction
+	super NitAction
+
+	# Model to explore and render.
+	var model: Model
+
+	# Modelbuilder used to access sources.
+	var modelbuilder: ModelBuilder
+
+	# TODO handle more than full namespaces.
+	redef fun answer(request, url) do
+		var namespace = request.param("namespace")
+		if namespace == null or namespace.is_empty then
+			return render_error(400, "Missing :namespace.")
+		end
+		var mentities = model.collect_by_namespace(namespace)
+		if mentities.is_empty then
+			return render_error(404, "No mentity matching this namespace.")
+		end
+		var view = new HtmlDocPage(modelbuilder, mentities.first)
+		return render_view(view)
+	end
+end
+
+# Return a random list of MEntities.
+class RandomAction
+	super NitAction
+
+	# Model to explore and render.
+	var model: Model
+
+	# TODO handle more than full namespaces.
+	redef fun answer(request, url) do
+		var n = request.int_arg("n") or else 10
+		var k = request.string_arg("k") or else "modules"
+		var mentities: Array[MEntity]
+		if k == "modules" then
+			mentities = model.mmodules.to_a
+		else if k == "mclassdefs" then
+			mentities = new Array[MClassDef]
+			for mclass in model.mclasses do
+				mentities.add_all(mclass.mclassdefs)
+			end
+		else
+			mentities = new Array[MPropDef]
+			for mprop in model.mproperties do
+				mentities.add_all(mprop.mpropdefs)
+			end
+		end
+		mentities.shuffle
+		mentities = mentities.sub(0, n)
+		if request.bool_arg("json") or else false then
+			var json = new JsonArray
+			for mentity in mentities do
+				json.add mentity.to_json
+			end
+			return render_json(json)
```

Comment:
Yet another json case. Maybe provide a `NitAction::render_json_entities(mentities: Collection[MEntity])` that factorize most of the code?